chirho/explainable/handlers/components.py

```diff
@@ -8,12 +8,47 @@
 from chirho.counterfactual.handlers.selection import get_factual_indices
 from chirho.explainable.internals import uniform_proposal
 from chirho.indexed.ops import IndexSet, gather, indices_of, scatter_n
-from chirho.observational.handlers import soft_neq
+from chirho.observational.handlers import soft_eq, soft_neq
 
 S = TypeVar("S")
 T = TypeVar("T")
 
 
+def sufficiency_intervention(
+    support: constraints.Constraint,
+) -> Callable[[torch.Tensor], torch.Tensor]:
+    """
+    Creates a sufficiency intervention for a single sample site, determined by
+    the site name, intervening to keep the value as in the factual world.
+
+    :param support: The support constraint for the site.
+    :param name: The sample site name.
+
+    :return: A function that takes a `torch.Tensor` as input
+        and returns the factual value at the named site as a tensor.
+
+    Example::
+
+        >>> with MultiWorldCounterfactual() as mwc:
+        >>>     value = pyro.sample("value", proposal_dist)
+        >>>     intervention = sufficiency_intervention(support)
+        >>>     value = intervene(value, intervention)
+    """
+
+    def _sufficiency_intervention(value: torch.Tensor) -> torch.Tensor:
+
+        indices = indices_of(value)
+
+        factual_value = gather(
+            value,
+            IndexSet(**{index: {0} for index in indices}),
```

Comment:
I think this depends on whether we want `sufficiency_intervention` to take the same arguments are `random_intervention` right now, or whether you want to have different syntactic behavior, or whether you want to revise both so that they require antecedents and then we move some of what the handler does right now in the beginning within the scope of these. Give it a thought and let me know.

Reply:
See my comment on `sufficiency_intervention`.

Reply:
it's not clear whether the remark applies to `random_intervention` as well. Seems like it does, can you confirm?

Reply:
As in, do you want the work that precedes random intervention in the handler to connect `random_intervention` with `antecedents` to be now performed within `random_intervention`?

Reply:
`random_intervention` does not do computations that depend on or modify the index variables of its input, so I don't see why you'd need to change it here?